### DISCUSSION TITLE
Welcome to exer_log Discussions!

### DISCUSSION THREAD

**Discussion by KalleHallden** [EXERLOG/exer_log#67] [Announcements]
## Welcome!
  We’re using Discussions as a place to connect with each other and for discussions relating to the project.

  ➡️ You can **convert issues to discussions** either individually or bulk by labels.

**Comment by momshaddinury** [comment]
This is great. Now we can freely discuss stuff.

**Comment by suzanpradhan** [comment]
I guess we need to create a discussion for whole UI update like colors, assets, layouts etc. What do you say?

**Reply by momshaddinury** [reply]
Regarding assets, files are going to stay [here](https://github.com/KalleHallden/exer_log/tree/master/assets) and in-code references for them will be [here](https://github.com/KalleHallden/exer_log/blob/master/lib/src/utils/assets.dart)

**Reply by momshaddinury** [reply]
Regarding colors, usually, I create a class named `AppColors` and hold `static const` instances of the colors there. In the code, I just call `AppColors.primary`.

And, regarding naming color variables it's better to name them according to their color name. 
Example:
```
class AppColors {
  /// DO
  static const red = Colors.red;
  /// DON'T DO
  static const appBarBackground = Colors.red;
}
```

**Reply by KalleHallden** [reply]
Yeah great idea, just not sure about which way to go. I guess the app colors class might be a good way to go.

**Reply by momshaddinury** [reply]
@ckelwin @jorre127 any suggestions?

**Reply by jorre127** [reply]
@dinurymomshad @KalleHallden @ckelwin I usually have a file called ThemeColors that stores the actual colours, almost exactly the same like the example you gave above here. But I don't reference these colours directly in the code. Instead I have a class between this called AppTheme or something that can house a little more specific names like for example appBarBackground like you said above. And this will reference the ThemeColors. I also usually have 2 instances 1 for dark mode and 1 for light mode so it's really easy to add light/dark mode in the future. This would look something like this:

```
class ThemeColors {
  ThemeColors._();

  static const primary = //primary color;
  static const accent = //accent;
  static const red = //red;
}
```

the appTheme: 

```
class ColorTheme {
  final Color primary;
  final Color accent;
  final Color background;
  final Color disabled;
  final Color lightIcon;
  final Color darkIcon;
  final Color lightProgressIndicator;
  final Color darkProgressIndicator;
}

class AppTheme{
 static const _lightTheme = ColorTheme(primary: ThemeColors.primary, lightIcon: ThemeColors.white, darkIcon: ThemeColors.grey etc.);
 static const _darkTheme = ColorTheme(//put in dark colours here);

 AppTheme of(BuildContext context){
 // check if we're light or dark mode
  return _lightTheme or _darkTheme
 }
}
```

We can then use it in the app like:

```
final theme = AppTheme.of(context);
theme.primary, theme.darkIcon etc.
```

We could also write a widget which passes the theme to its builder so we don't have to write the final theme = AppTheme.of(context) everytime.

**Reply by momshaddinury** [reply]
@jorre127 Do you want to cover that task? It seems like you have a better idea than me about this.

**Reply by jorre127** [reply]
@dinurymomshad Jup that's fine by me, can we convert this discussion into a issue? or do we have to create a separate issue.

**Reply by momshaddinury** [reply]
@jorre127 A separate issue since you will only focus on `ThemeColors`. I suggest adding the files inside the `core/theme` directory.

**Reply by suzanpradhan** [reply]
I have created separate discussion just for UI/UX #87 so we can discuss everything related to it then create issues based on the topic.

**Reply by jorre127** [reply]
@dinurymomshad @suzanpradhan @KalleHallden  Implemented the things we discussed in PR #94

**Comment by thefr0gg** [comment]
i have found an overflow in the [calendar_widget.dart](https://github.com/KalleHallden/exer_log/blob/alpha/lib/UI/calendar/widgets/calendar_widget.dart)
***
![Screenshot_20220818-150554](https://user-images.githubusercontent.com/86024158/185415599-5a21f741-b0b3-40df-abb0-fd0b0cf8d467.png)
***
and we can fix it by changing the line 43 from 
``` dart 
height: screenHeight * 0.4
```
to 
``` dart
height: screenHeight * 0.43, 
```
it not somthing big but it may affect the UI.

**Reply by momshaddinury** [reply]
I am guessing the issue is somewhere else. Judging by the gap between the Calendar widget and the button it should not overflow.

**Reply by jorre127** [reply]
I think it’s because it’s based on a percentage of the screensize. It’s probably better to just size the calendar based on its content instead.

**Reply by thefr0gg** [reply]
yeah you are right

**Comment by jorre127** [comment]
@KalleHallden I don't know how you would implement this on GitHub, but maybe we should implement a system where a PR has to at least be reviewed by a certain number of people, or a certain group of people, before being merged. Because now some PR's are being merged with some mistakes (pr #99 for example), because nobody has reviewed them at the time. This could create an endless cycle of PR's fixing the code, but then a PR gets merged that doesn't follow the guidelines and then it needs to get fixed again.

**Reply by momshaddinury** [reply]
Another bug was introduced by #122. We should really focus on adding maintainers. Asking people to add videos/images of before and after would also help. @KalleHallden

**Reply by KalleHallden** [reply]
Yeah I agree I think this would be very helpful.

**Reply by momshaddinury** [reply]
@KalleHallden you need to decide who becomes maintainers. And nothing should be merged without testing by pulling the changes in maintainers local and then verifying if everything works.

**Reply by momshaddinury** [reply]
Also voting for a discord server maintainers channel or something for more convenient communication. I think you already have a server.
